google/cloud/aiplatform/training_jobs.py

```diff
@@ -3346,3 +3347,255 @@ def _run(
+                applicable).
+            sentiment_max (int):
+                Required and only applicable for sentiment task. A sentiment is expressed as an integer
+                ordinal, where higher value means a more
+                positive sentiment. The range of sentiments that
+                will be used is between 0 and sentimentMax
+                (inclusive on both ends), and all the values in
+                the range must be represented in the dataset
+                before a model can be created.
+                Only the Annotations with this sentimentMax will
+                be used for training. sentimentMax value must be
+                between 1 and 10 (inclusive).
+            project (str):
+                Optional. Project to run training in. Overrides project set in aiplatform.init.
+            location (str):
+                Optional. Location to run training in. Overrides location set in aiplatform.init.
+            credentials (auth_credentials.Credentials):
+                Optional. Custom credentials to use to run call training service. Overrides
+                credentials set in aiplatform.init.
+        """
+        super().__init__(
+            display_name=display_name,
+            project=project,
+            location=location,
+            credentials=credentials,
+        )
+        self._prediction_type = prediction_type
+        self._multi_label = multi_label
+        self._sentiment_max = sentiment_max
+        self._model_hint = model_hint
+
+        training_task_definition: str
+        training_task_inputs_dict: proto.Message
+
+        if self._prediction_type == "classification":
+            training_task_definition = (
+                schema.training_job.definition.automl_text_classification_task
+            )
+
+            if self._multi_label is None:
+                raise ValueError(
+                    "The multi_label parameter must be provided for a prediction_type of 'classification'"
+                )
+
+            training_task_inputs_dict = AutoMlTextClassificationInputs(
+                multi_label=self._multi_label
+            ).to_value()
+        elif self._prediction_type == "extraction":
+            training_task_definition = (
+                schema.training_job.definition.automl_text_extraction_task
+            )
+
+            training_task_inputs_dict = AutoMlTextExtractionInputs().to_value()
+        elif self._prediction_type == "sentiment":
+            training_task_definition = (
+                schema.training_job.definition.automl_text_sentiment_task
+            )
+
+            if self._sentiment_max is None:
+                raise ValueError(
+                    "Tge sentiment_max parameter must be provided for a prediction_type of 'sentiment'"
+                )
+
+            training_task_inputs_dict = AutoMlTextSentimentInputs(
+                sentiment_max=self._sentiment_max
+            ).to_value()
+        else:
```

Comment:
We should check for this first and exit early if it's not one of the required types.

Reply:
Not sure I understand correctly, but I can move this check higher up in the function. I'd refrain from adding an additional check that repeats this logic (to reduce potential for bugs in the future if someone forgets to sync that logic with this if-else statement).